app/src/main/java/org/mozilla/fenix/home/sessioncontrol/SessionControlUIView.kt

```diff
@@ -152,10 +151,8 @@ class SessionControlUIView(
     }
 
     override fun updateView() = Consumer<SessionControlState> {
-        // Workaround for list not updating until scroll on Android 5 + 6
-        if (Build.VERSION.SDK_INT < Build.VERSION_CODES.N) {
-            sessionControlAdapter.submitList(null)
-        }
+        // Empty list before adding items specific to type of new session. See https://github.com/mozilla-mobile/fenix/issues/6516
+        sessionControlAdapter.submitList(null)
```

Comment:
Could this only be done when the mode changes? We don't want to refresh all the views when making small changes like opening a new tab.

Reply:
I was just working on this change. :) 

Slightly related to this: Do you know if any diff-ing is made against opened tabs when you open tabs from history of bookmarks? It seems it opens again some pages you already have in open tabs list.

Reply:
As far as I know, we always open new tabs for bookmarks/history.